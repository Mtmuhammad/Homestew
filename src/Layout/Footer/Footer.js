import React from "react";
import "./Footer.scss";
import FooterBox from "../FooterBox/FooterBox";
import Instagram1 from "../../Assets/Images/instagram/Instagram1.jpg"
import Instagram2 from "../../Assets/Images/instagram/Instagram2.jpg"
import Instagram3 from "../../Assets/Images/instagram/Instagram3.jpg"
import Instagram4 from "../../Assets/Images/instagram/Instagram4.jpg"
import Instagram5 from "../../Assets/Images/instagram/Instagram5.jpg"
import Instagram6 from "../../Assets/Images/instagram/Instagram6.jpg"
import avatar from "../../Assets/Images/instagram/avatar.jpg"

const Footer = () => {
  return (
    <footer className="footer">
      <div className="container-fluid">
        <div className="footer-instagram">
          <div className="footer-instagram-avatar">
            <div className="footer-instagram-avatar-pic">
              <img
                src={avatar}
                alt="Instagram Avatar"
              />
            </div>
            <div className="footer-instagram-avatar-text">
              <h5>@homestew</h5>
              <span>22,5k followers</span>
            </div>
          </div>
          <div className="row">
            <FooterBox img={Instagram1} />
            <FooterBox img={Instagram2} />
            <FooterBox img={Instagram3} />
            <FooterBox img={Instagram4} />
            <FooterBox img={Instagram5} />
            <FooterBox img={Instagram6} />
          </div>
        </div>
      </div>
      <div className="container">
        <div className="row">
          <div className="col-lg-12">
            <div className="footer-text">
              <div className="footer-logo">
                <a href="/">
                  <img
                    className="img-fluid mb-4"
                    src="./Homestew-alt-logo.svg"
                    alt="Footer alt logo"
                  />
                </a>
              </div>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua. Quis
                ipsum suspendisse ultrices gravida. Risus commodo viverra
                maecenas accumsan lacus vel facilisis.
              </p>
              <div className="footer-social mb-5">
                <a href="#0">
                  <i data-testid="footer-icon" className="fa fa-facebook"></i>
                </a>
                <a href="#0">
                  <i data-testid="footer-icon" className="fa fa-twitter"></i>
                </a>
                <a href="#0">
                  <i data-testid="footer-icon" className="fa fa-instagram"></i>
                </a>
                <a href="#0">
                  <i data-testid="footer-icon" className="fa fa-youtube-play"
                  ></i>
                </a>
                <a href="#0">
                  <i data-testid="footer-icon" className="fa fa-envelope-o"></i>
                </a>
              </div>
              <div className="footer-copyright">
                <p className="mb-0">Copyright Â© 2022 All rights reserved</p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
