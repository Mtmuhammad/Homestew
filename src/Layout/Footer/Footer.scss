@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

.footer {
  background: $light-primary-green;
  padding-top: 20px;

  &-copyright {
    font-size: 1.5rem;
    border-top: 1px solid #e1e1e1;
    text-align: center;
    padding: 30px 0 25px;

    @include respond(xsphone) {
      padding: 20px 0 15px;
    }

    & p {
      font-size: 1.4rem;
      font-weight: 400;
      line-height: 26px;
      margin: 0 0 15px 0;
    }
  }

  &-social a {
    color: $primary-font;
    display: inline-block;
    font-size: 1.6rem;
    height: 46px;
    width: 46px;
    background: #e7e7e7;
    border-radius: 50%;
    line-height: 46px;
    text-align: center;
    transition: all, 0.3s;
    margin-right: 6px;

    &:hover {
      background: $primary-green;
      color: white;
    }
  }

  &-logo {
    margin-bottom: 10px;

    & img {
      height: 8rem;
    }
  }

  &-instagram {
    padding: 0 15px;
    position: relative;

    &-avatar {
      display: inline-block;
      padding: 20px 30px;
      background: white;
      position: absolute;
      height: 90px;
      width: 240px;
      left: 50%;
      top: 50%;
      transform: translate(-120px, -50px);
      z-index: 1;

      @include respond(port) {
        padding: 20px;
      }
      @include respond(xsphone) {
        height: 70px;
      }

      &-pic {
        float: left;
        margin-right: 15px;

        & img {
          height: 5rem;
          width: 5rem;
          border-radius: 50%;
        }
      }

      &-text {
        overflow: hidden;

        & h5 {
          font-size: 2.2rem;
          font-weight: 700;
          margin-bottom: 5px;
        }

        & span {
          font-size: 1.6rem;
          color: $icon-hover-grey;
        }
      }
    }
  }

  &-text {
    padding-top: 50px;
    text-align: center;

    @include respond(xsphone) {
      padding-top: 25px;
    }

    & p {
      font-size: 1.7rem;
      margin-bottom: 30px;
      padding: 0 26rem;

      @include respond(land) {
        padding: 0 20rem;
      }

      @include respond(port) {
        padding: 0 10rem;
      }
      @include respond(sport) {
        padding: 0;
      }
    }
  }
}
