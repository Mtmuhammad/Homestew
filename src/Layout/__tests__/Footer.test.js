import React from "react";
import { render, screen } from "@testing-library/react";
import Footer from "../Footer/Footer";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<Footer />);
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("render footer", () => {
   test("should render footer avatar", () => {
      render(<Footer/>)
      const avatar = screen.getByAltText(/instagram avatar/i)
      expect(avatar).toBeInTheDocument()

      const avatarHeader = screen.getByRole("heading", {name: "@homestew"})
      expect(avatarHeader).toBeInTheDocument()
   })

   test("should render images", () => {
      render(<Footer/>)
      const images = screen.getAllByAltText(/instagram post/i)
      expect(images.length).toBe(6)
   })

   test("should render footer logo", () => {
      render(<Footer/>)
      const footerLogo = screen.getByAltText(/footer alt logo/i)
      expect(footerLogo).toBeInTheDocument()
      expect(footerLogo).toHaveClass("img-fluid")
   })

   test("should render social icons", () => {
      render(<Footer/>)
      const footerIcons = screen.getAllByTestId("footer-icon")
      expect(footerIcons.length).toBe(5)
   })
})