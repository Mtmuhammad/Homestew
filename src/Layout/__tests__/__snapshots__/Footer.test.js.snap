// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container-fluid"
    >
      <div
        class="footer-instagram"
      >
        <div
          class="footer-instagram-avatar"
        >
          <div
            class="footer-instagram-avatar-pic"
          >
            <img
              alt="Instagram Avatar"
              src="avatar.jpg"
            />
          </div>
          <div
            class="footer-instagram-avatar-text"
          >
            <h5>
              @homestew
            </h5>
            <span>
              22,5k followers
            </span>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram1.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram2.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram3.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram4.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram5.jpg"
            />
          </div>
          <div
            class="col-6 col-sm-4 col-md-2"
          >
            <img
              alt="Instagram post"
              class="img-fluid instagram-post"
              src="Instagram6.jpg"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <div
            class="footer-text"
          >
            <div
              class="footer-logo"
            >
              <a
                href="/"
              >
                <img
                  alt="Footer alt logo"
                  class="img-fluid mb-4"
                  src="./Homestew-alt-logo.svg"
                />
              </a>
            </div>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Quis ipsum suspendisse ultrices gravida. Risus commodo viverra maecenas accumsan lacus vel facilisis.
            </p>
            <div
              class="footer-social mb-5"
            >
              <a
                href="#0"
              >
                <i
                  class="fa fa-facebook"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-twitter"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-instagram"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-youtube-play"
                  data-testid="footer-icon"
                />
              </a>
              <a
                href="#0"
              >
                <i
                  class="fa fa-envelope-o"
                  data-testid="footer-icon"
                />
              </a>
            </div>
            <div
              class="footer-copyright"
            >
              <p
                class="mb-0"
              >
                Copyright Â© 2022 All rights reserved
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
